### DISCUSSION TITLE
bootloop when trying to install legacy update

### DISCUSSION THREAD

**Discussion by IMat14** [LegacyUpdate/LegacyUpdate#312] [Questions & Help]
Hello,
I've been trying to install legacy update on win7 pc, but when it reboot for the first time, it says a thing like "the pc is getting ready for the first time" and then it just reboot and that repeat itself. How do i do to continue the setup dans to stop the bootloop ?

**Comment by kirb** [comment]
Sorry for the issue. This is being tracked in issue #285 currently.

This happens when there’s something stopping the updates from finishing installation. Legacy Update is trying to reboot to let Windows finish installation, but it never does finish.

Since I’m trying to figure out what causes this, could I ask how you installed Windows (what DVD/iso did you use), and if maybe there were already some updates installed on it before you ran Legacy Update?

---

edit: Updating here because this is still coming up fairly often. I believe this is specifically affecting French and Russian installs of Windows 7. (If you use a different language and still ran into this, please let me know.)

# How to get out of the reboot loop

This workaround is kinda annoying, but it can be done:

1. Start up the PC and hold F8, select Startup Repair. Or, boot from your Windows DVD and click the Startup Repair option.
3. Select your Windows drive, then Command Prompt
4. Type `regedit` and press Enter
5. Click HKEY_LOCAL_MACHINE, then click File &rarr; Load Hive
6. Type `C:\Windows\system32\config\system`
7. Give it any name, doesn’t matter what you use here
8. Open the key you just named &rarr; Setup
9. Double-click SetupType and change it to `0`
10. Close all windows and select to restart.

If it hasn’t scared you off, you can try running Legacy Update again after you get back to the desktop. The updates that needed installing have already been installed, so no more reboots are needed.

---

Absolutely not required, but if you have a bit more patience and want to help out, I’d appreciate if you could do a few more steps before you close regedit:

1. Repeat steps 5-7, but select the file `software` instead of `system`
2. Go to your key &rarr; Microsoft\Windows\CurrentVersion\Component Based Servicing
3. If you see PendingPackages here, click it &rarr; File &rarr; Export

You might want to save it somewhere like `C:\Users\YourName\Desktop`. Then when you’re back at your desktop, upload that .reg here.

**Reply by IMat14** [reply]
Thanks for the reply,
I installed windows with a usb stick and a x64 .iso.
Just done it and it works, thanks again.
And for the .reg file, i guess the "PendingPackages" and the "PackagesPending" are the same so here is the .reg file:
[regedit thing.zip](https://github.com/user-attachments/files/18275687/regedit.thing.zip)

**Reply by kirb** [reply]
Thanks so much for doing that! What was the filename of the .iso (or sha1, if you know how to find that)? The ones I’ve tested on worked fine, which has been kinda annoying 😅

**Reply by IMat14** [reply]
No problem !
the filename was : fr_windows_7_home_premium_with_sp1_x64_dvd_u_676521.iso

**Reply by kirb** [reply]
Whoa, interesting. Currently on Discord I'm discussing this with someone who used this exact iso but in Ultimate (fr_windows_7_ultimate_with_sp1_x64_dvd_u_677299.iso) and reproduced the same bootloop. Downloading that iso now to see if I can reproduce it...

**Reply by Teligin** [reply]
Hi! Haven't checked this in a while, i see you need my iso's so here are their names:
Windows 7 SP1 x86-x64 RU_AIO
ru_windows_8.1_professional
ru_windows_vista_with_sp2_x64
BUT i think i changed those names, so, if you need, i can upload these to my icloud drive or send them in telegram. If you want telegram contact me @stevedjobs if icloud just reply here, ill see an email from github)

**Reply by moeenio** [reply]
> Type C:\Windows\system32\config\system

fwiw, here i had to use `D:`, looked in "my computer" to  figure out based on the size of that storage

**Comment by mobluse** [comment]
I have this bootloop problem in Windows Vista (see https://github.com/LegacyUpdate/LegacyUpdate/issues/307#issuecomment-2586044844 and https://github.com/LegacyUpdate/LegacyUpdate/issues/285#issuecomment-2590265910 ), but I can't get into Windows where I can run programs like regedit. I can boot into Linux on the same machine and run a registry editor for Linux, but I'm not sure how to proceed. This works:
```
$ sudo mount /dev/sda2 /mnt
$ cd /mnt/Windows/System32/config/
$ chntpw -l SAM | iconv -f cp1252 -t utf8
Hive <SAM> name (from header): <\SystemRoot\System32\Config\SAM>
ROOT KEY at offset: 0x001020 * Subkey indexing type is: 666c <lf>
File size 262144 [40000] bytes, containing 6 pages (+ 1 headerpage)
Used for data: 233/51376 blocks/bytes, unused: 12/5776 blocks/bytes.


* SAM policy limits:
Failed logins before lockout is: 0
Minimum password length        : 0
Password history count         : 0
| RID -|---------- Username ------------| Admin? |- Lock? --|
| 01f4 | Administratör                  | ADMIN  | dis/lock |
| 01f5 | Gäst                           |        | dis/lock |
| 03eb | Mikael                         | ADMIN  | dis/lock |
$ chntpw -N -e SAM
chntpw version 0.99.6 110511 , (c) Petter N Hagen
Hive <SAM> name (from header): <\SystemRoot\System32\Config\SAM>
ROOT KEY at offset: 0x001020 * Subkey indexing type is: 666c <lf>
File size 262144 [40000] bytes, containing 6 pages (+ 1 headerpage)
Used for data: 233/51376 blocks/bytes, unused: 12/5776 blocks/bytes.


* SAM policy limits:
Failed logins before lockout is: 0
Minimum password length        : 0
Password history count         : 0
Simple registry editor. ? for help.

> ?
Simple registry editor:
hive [<n>]             - list loaded hives or switch to hive numer n
cd <key>               - change current key
ls | dir [<key>]       - show subkeys & values,
cat | type <value>     - show key value
dpi <value>            - show decoded DigitalProductId value
hex <value>            - hexdump of value data
ck [<keyname>]         - Show keys class data, if it has any
nk <keyname>           - add key
dk <keyname>           - delete key (must be empty)
ed <value>             - Edit value
nv <type#> <valuename> - Add value
dv <valuename>         - Delete value
delallv                - Delete all values in current key
rdel <keyname>         - Recursively delete key & subkeys
ek <filename> <prefix> <keyname>  - export key to <filename> (Windows .reg file format)
debug                  - enter buffer hexeditor
st [<hexaddr>]         - debug function: show struct info
q                      - quit

> q

Hives that have changed:
 #  Name
None!

$ chntpw -N -e SYSTEM
chntpw version 0.99.6 110511 , (c) Petter N Hagen
Hive <SYSTEM> name (from header): <SYSTEM>
ROOT KEY at offset: 0x001020 * Subkey indexing type is: 686c <lh>
File size 29360128 [1c00000] bytes, containing 5809 pages (+ 1 headerpage)
Used for data: 460169/28340576 blocks/bytes, unused: 16810/665728 blocks/bytes.

Simple registry editor. ? for help.

> 
```
I was wondering if you could help me to break the bootloop when I boot into Windows in a safe way, i.e. without risking the Vista installation.

**Reply by kirb** [reply]
Can you boot from Startup Repair in the F8 menu, or use a Vista DVD? 

I’m not the most familiar with chntpw so I can’t give you the exact commands, but you need to load the SYSTEM hive, then under the Setup key (which is directly in the root of the hive), set DWORD value SetupType to 0.

**Reply by mobluse** [reply]
Thanks! I got my Vista computer to boot normally after doing this from Linux and then rebooting to Windows:
```
user@computer:/mnt/Windows/System32/config$ chntpw -N -e SYSTEM
chntpw version 0.99.6 110511 , (c) Petter N Hagen
Hive <SYSTEM> name (from header): <SYSTEM>
ROOT KEY at offset: 0x001020 * Subkey indexing type is: 686c <lh>
File size 29360128 [1c00000] bytes, containing 5809 pages (+ 1 headerpage)
Used for data: 460169/28340576 blocks/bytes, unused: 16810/665728 blocks/bytes.

Simple registry editor. ? for help.

> ?
Simple registry editor:
hive [<n>]             - list loaded hives or switch to hive numer n
cd <key>               - change current key
ls | dir [<key>]       - show subkeys & values,
cat | type <value>     - show key value
dpi <value>            - show decoded DigitalProductId value
hex <value>            - hexdump of value data
ck [<keyname>]         - Show keys class data, if it has any
nk <keyname>           - add key
dk <keyname>           - delete key (must be empty)
ed <value>             - Edit value
nv <type#> <valuename> - Add value
dv <valuename>         - Delete value
delallv                - Delete all values in current key
rdel <keyname>         - Recursively delete key & subkeys
ek <filename> <prefix> <keyname>  - export key to <filename> (Windows .reg file format)
debug                  - enter buffer hexeditor
st [<hexaddr>]         - debug function: show struct info
q                      - quit

> cd Setup

\Setup> ls
Node has 5 subkeys and 10 values
  key name
  <AllowStart>
  <Pid>
  <Service Reporting API>
  <SetupCL>
  <Status>
  size     type            value name             [value if type DWORD]
     4  REG_SZ            <OsLoaderPath>
     4  REG_DWORD         <RestartSetup>             0 [0x0]
     4  REG_DWORD         <SetupType>                2 [0x2]
    48  REG_SZ            <SystemPartition>
     4  REG_DWORD         <SystemSetupInProgress>    0 [0x0]
   114  REG_SZ            <CmdLine>
     4  REG_DWORD         <SetupPhase>               0 [0x0]
     4  REG_DWORD         <OOBEInProgress>           0 [0x0]
    38  REG_SZ            <WorkingDirectory>
    52  REG_MULTI_SZ      <CloneTag>

\Setup> ed SetupType
EDIT: <SetupType> of type REG_DWORD with length 4 [0x4]
DWORD: Old value 2 [0x2], enter new value (prepend 0x if hex, empty to keep old value)
-> 0
DWORD: New value 0 [0x0], 
\Setup>     

\Setup> ls
Node has 5 subkeys and 10 values
  key name
  <AllowStart>
  <Pid>
  <Service Reporting API>
  <SetupCL>
  <Status>
  size     type            value name             [value if type DWORD]
     4  REG_SZ            <OsLoaderPath>
     4  REG_DWORD         <RestartSetup>             0 [0x0]
     4  REG_DWORD         <SetupType>                0 [0x0]
    48  REG_SZ            <SystemPartition>
     4  REG_DWORD         <SystemSetupInProgress>    0 [0x0]
   114  REG_SZ            <CmdLine>
     4  REG_DWORD         <SetupPhase>               0 [0x0]
     4  REG_DWORD         <OOBEInProgress>           0 [0x0]
    38  REG_SZ            <WorkingDirectory>
    52  REG_MULTI_SZ      <CloneTag>

\Setup> q

Hives that have changed:
 #  Name
 0  <SYSTEM>
Write hive files? (y/n) [n] : y
 0  <SYSTEM> - OK
```

**Comment by moeenio** [comment]
I too had the same problem with an iso named `fr_windows_7_ultimate_with_sp1_x64_dvd_u_677299.iso`, with sha1sum `5370bb44f497e1b19114c7cf70f5cdef9f3a2582`. Here is my [PendingPackages.reg](https://bin.siick.fr/?cf892dbffa17dbee#FGngADUaFg1yrnUeqvYFLhzTNh5Ldbae3k1qKdwaG6uS)
